bundles/org.openhab.binding.ambientweather/src/main/java/org/openhab/binding/ambientweather/internal/processor/Ws0900ipProcessor.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ambientweather.internal.processor;
+
+import static org.openhab.binding.ambientweather.internal.AmbientWeatherBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.ImperialUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.openhab.binding.ambientweather.internal.handler.AmbientWeatherStationHandler;
+import org.openhab.binding.ambientweather.internal.model.EventDataJson;
+import org.openhab.binding.ambientweather.internal.util.PressureTrend;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link Ws0900ipProcessor} is responsible for updating
+ * the channels associated with the WS-0900-IP weather stations in
+ * response to the receipt of a weather data update from the Ambient
+ * Weather real-time API.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class Ws0900ipProcessor extends AbstractProcessor {
+
+    // Used to calculate barometric pressure trend
+    private PressureTrend pressureTrend = new PressureTrend();
+
+    @Override
+    public void processInfoUpdate(AmbientWeatherStationHandler handler, String station, String name, String location) {
+        // Update name and location channels
+        logger.debug("Station {}: Updating station information channels", station);
+        handler.updateChannel(CHGRP_STATION + "#" + CH_NAME, new StringType(name));
+        handler.updateChannel(CHGRP_STATION + "#" + CH_LOCATION, new StringType(location));
+    }
+
+    @Override
+    public void processWeatherData(AmbientWeatherStationHandler handler, String station, String jsonData) {
+        try {
+            logger.debug("Station {}: Updating weather data channels", station);
```

Comment:
Try to make try-catch blocks as short as possible. Let it only wrap the json part and do the channel updates after

Reply:
Done for all processors.